changelog-endpoint/src/integration-test/kotlin/endpoint/ChangelogModuleIntegrationTest.kt

```diff
@@ -0,0 +1,133 @@
+
+    @AfterAll
+    fun tearDown() {
+        changelogEnvironment.close()
+    }
+
+    /**
+     * @given script file with Iroha account creation logic
+     * @when script file is passed to changelog service
+     * @then account is created
+     */
+    @Test
+    fun testExecuteFile() = withTestApplication({
+        changelogModule(changelogEnvironment.changelogExecutor)
+    }) {
+        val random = Random()
+        val randomAccountName = random.nextInt().absoluteValue.toString()
+        val randomSchema = random.nextInt().absoluteValue.toString()
+        //Create file that creates randomly named account
+        changelogEnvironment.createAccountScriptFile(randomAccountName, randomSchema)
+        val request = changelogEnvironment.createChangelogFileRequest(randomAccountName)
+        with(handleRequest(HttpMethod.Post, "/changelog/changelogFile") {
+            setBody(changelogEnvironment.gson.toJson(request))
+        }) {
+            assertEquals(HttpStatusCode.OK, response.status())
+        }
+        val changelogReducedTxHash = JSONObject(
+            changelogEnvironment.queryAPI.getAccountDetails(
+                changelogHistoryStorageAccountId,
+                ChangelogInterface.superuserAccountId,
+                randomSchema
+            )
+        ).getJSONObject(ChangelogInterface.superuserAccountId).get(randomSchema).toString()
+        assertNotNull(changelogReducedTxHash)
+
+        val account = changelogEnvironment.queryAPI.getAccount("$randomAccountName@d3").account
+        assertEquals(
+            "{\"superuser@bootstrap\": {\"test_key\": \"test_value\"}}",
+            account.jsonData
+        )
+    }
+
+    /**
+     * @given script file with Iroha account creation logic
+     * @when script file is passed to changelog service twice
+     * @then account is created, second changelog attempt didn't make any effect
+     */
+    @Test
+    fun testIdempotent() = withTestApplication({
+        changelogModule(changelogEnvironment.changelogExecutor)
+    }) {
+        val random = Random()
+        val randomAccountName = random.nextInt().absoluteValue.toString()
+        val randomSchema = random.nextInt().absoluteValue.toString()
+        //Create file that creates randomly named account
+        changelogEnvironment.createAccountScriptFile(randomAccountName, randomSchema)
+        val request = changelogEnvironment.createChangelogFileRequest(randomAccountName)
+
+        with(handleRequest(HttpMethod.Post, "/changelog/changelogFile") {
+            setBody(changelogEnvironment.gson.toJson(request))
+        }) {
+            assertEquals(HttpStatusCode.OK, response.status())
+        }
+        //Second changelog attempt
+        with(handleRequest(HttpMethod.Post, "/changelog/changelogFile") {
+            setBody(changelogEnvironment.gson.toJson(request))
+        }) {
+            assertEquals(HttpStatusCode.OK, response.status())
```

Comment:
status is ok but how can we guarantee that nothing changed?

Reply:
It's a good point. I will modify the test.